src/main/java/rx/internal/operators/OperatorSwitchIfEmpty.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.operators;
+
+import rx.Observable;
+import rx.Producer;
+import rx.Subscriber;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * If the Observable completes without emitting any items, subscribe to an alternate Observable. Allows for similar
+ * functionality to {@link rx.internal.operators.OperatorDefaultIfEmpty} except instead of one item being emitted when
+ * empty, the results of the given Observable will be emitted.
+ */
+public class OperatorSwitchIfEmpty<T> implements Observable.Operator<T, T> {
+    private final Observable<T> alternate;
+
+    public OperatorSwitchIfEmpty(Observable<T> alternate) {
+        this.alternate = alternate;
+    }
+
+    @Override
+    public Subscriber<? super T> call(Subscriber<? super T> child) {
+        final SwitchIfEmptySubscriber parent = new SwitchIfEmptySubscriber(child);
+        child.add(parent);
```

Comment:
The problem with directly adding this to the child is that the child will retain a reference to a now-dead switcher. Instead, I suggest having a ```SerialSubscription```:

```java
SerialSubscription ssub = new SerialSubscription();
SwitchIfEmptySubscriber parent = new SwitchIfEmptySubscriber(child, ssub);
ssub.add(parent);
child.add(ssub);
return parent;
```
The ```subscribeToAlternate``` should now start with:
```java
ssub.set(alternate.unsafeSubscribe(new Subscriber<T>() {
```